### PR TITLE
10.1.5 prevoker

### PR DESCRIPTION
**Pull request by trevorm4**
### Description

Adds Regenerative Magic module and updates exhil burst

### DO NOT LAND UNTIL 10.1.5

### CODE REVIEW

**Review by emallson — APPROVED** [review]

File: src/analysis/retail/evoker/shared/modules/talents/RegenerativeMagic.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  protected statTracker!: StatTracker;
+
+  constructor(options: Options) {
+    super(options);
+    this.addEventListener(Events.heal.by(SELECTED_PLAYER).spell(SPELLS.LEECH), this.onHeal);
+  }
+
+  onHeal(event: HealEvent) {
+    const percentFromTalent = LEECH_PERCENT / this.statTracker.currentLeechPercentage;
+    this.totalHealing += Math.floor(event.amount * percentFromTalent);
```
Comment (vohrr) [inline src/analysis/retail/evoker/shared/modules/talents/RegenerativeMagic.tsx]: Any reason we're not using calculateEffectiveHealing here?

Comment (trevorm4) [reply src/analysis/retail/evoker/shared/modules/talents/RegenerativeMagic.tsx]: Yea I should calc increase and then do that prolly,  I'll change that.

File: src/interface/guide/components/MajorDefensives/AllCooldownUsagesList.tsx [hunk 2]
```diff
@@ -253,6 +253,7 @@ const CooldownUsage = ({
   const [selectedMit, setSelectedMit] = useState<number | undefined>();
   const castEfficiency = useAnalyzer(CastEfficiency)?.getCastEfficiencyForSpell(analyzer.spell);
   const possibleUses = castEfficiency?.maxCasts ?? 0;
+  console.log(analyzer.spell.name, possibleUses);
```
Comment (vohrr) [inline src/interface/guide/components/MajorDefensives/AllCooldownUsagesList.tsx]: delete

File: src/analysis/retail/evoker/shared/modules/talents/RegenerativeMagic.tsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  protected statTracker!: StatTracker;
+
+  constructor(options: Options) {
+    super(options);
+    this.addEventListener(Events.heal.by(SELECTED_PLAYER).spell(SPELLS.LEECH), this.onHeal);
+  }
+
+  onHeal(event: HealEvent) {
+    const increase =
+      this.statTracker.currentLeechPercentage /
```
Comment (vohrr) [inline src/analysis/retail/evoker/shared/modules/talents/RegenerativeMagic.tsx:30]: Toss a "1 - increase" in here, what you have is going to feed 1.[Increase] into the function
i.e   0.12 / (0.12 - 0.03%) = 1.33

**Review by vohrr — CHANGES_REQUESTED** [review]

**Review by vohrr — APPROVED** [review]

**Pull request merged**